core/lib/zksync_core/src/api_server/web3/mod.rs

```diff
@@ -579,6 +585,8 @@ impl FullApiParams {
         let closing_vm_barrier = vm_barrier.clone();
         let health_updater = Arc::new(health_updater);
```

Comment:
This creates an `Arc<Arc<_>>`, which is probably not what you need. One option is to change `health_updater` field type to `Option<HealthUpdater>` and take it out before consuming `self`. Not ideal, but IMO better than wrapping the field in `Arc`.

Reply:
Oh, I just forgot to remove this line. Of course, since we already have an `Arc`, we don't need to wrap it twice.
I'm not sure about `Option` though -- it would add a bunch of `unwrap`s, e.g. in places where we subscribe to get a health check object.